src/math_functions.cpp

```diff
@@ -0,0 +1,818 @@
+#include "math_functions.h"
+
+namespace openmc {
+
+//==============================================================================
+// Module constants.
+//==============================================================================
+
+// TODO: cmath::M_PI has 3 more digits precision than the Fortran constant we
+// use so for now we will reuse the Fortran constant until we are OK with
+// modifying test results
+extern "C" const double PI {3.1415926535898};
+
+extern "C" const double SQRT_PI {std::sqrt(PI)};
+
+//==============================================================================
+// NORMAL_PERCENTILE calculates the percentile of the standard normal
+// distribution with a specified probability level
+//==============================================================================
+
+double __attribute__ ((const)) normal_percentile_c(const double p) {
+  double z;
+  double q;
```

Comment:
Declare `q` later when it is first used.

Reply:
- [x]